grpc/proto/v1/properties.proto

```diff
@@ -26,11 +26,73 @@ message Value {
     string blob_value = 10;
     PhoneNumber phone_value = 11;
     google.protobuf.NullValue null_value = 12;
+    string text_value = 13;
   }
 }
 
 message ListValue {
-  repeated Value values = 1;
+  repeated Value values = 1 [deprecated = true];
+  oneof kind {
+    NumberValues number_values = 2;
+    BoolValues bool_values = 3;
+    ObjectValues object_values = 4;
+    DateValues date_values = 5;
+    UuidValues uuid_values = 6;
+    IntValues int_values = 7;
+    TextValues text_values = 8;
+  }
+}
+
+message NumberValues {
+  /**
+    * The values are stored as a byte array, where each 8 bytes represent a single float64 value.
+    * The byte array is stored in little-endian order using uint64 encoding.
+    */
+  bytes values = 1;
+}
+
+message TextValues {
+  /**
+    * The values are stored as a byte array, where each byte contains a single UTF-8 character.
+    * Individual text values are delimited by a ',' character within the overall byte array.
+    */
+  bytes values = 1;
+}
+
+message BoolValues {
```

Comment:
does it make a difference for bool values at all? 🤔

Reply:
Not sure, but I thought it would be best to be consistent here. If the bool encoding used the bits rather than the bytes then it would be faster, which I'll see if I can make work today

Reply:
I've decided to revert bools and strings to using `repeated`. Making an efficient encoding/decoding logic is particularly involved for these types and the biggest win for strings will be in the Python parsing of the dates and uuids more efficiently

Reply:
sounds good!